### ISSUE TITLE
jsx-sort-props reservedFirst conflicts with callbacksLast

### ISSUE THREAD

**Issue by jakezatecky** [yannickcr/eslint-plugin-react#1632]
It appears that PR #1340 introduced a change when `reservedFirst ` is enabled such that it conflicts with the option `callbacksLast`. When this option is enabled, ESLint wants callbacks to be sorted alphabetically with respect to the rest of the properties. The following will incorrectly generate an error:

`eslintrc`:

``` json
{
  "rules": {
    "react/jsx-sort-props": ["error", {
      "callbacksLast": true,
      "reservedFirst": true
    }]
  }
}
```

``` jsx
<button
  className={className}
  disabled={disabled}
  type="button"
  onClick={this.onClick}
>
  Submit
</button>
```

Notice that `onClick` occurs after `type`, which should be the case given the rule configuration. However, the change in this PR made it to where the React plugin wants `onClick` to appear before `type`, effectively disabling the entire `callbacksLast` rule.

I have confirmed that reverting `jsx-sort-props.js` fixes this bug, but obviously that is not ideal because the alphabetic sorting fix in that PR is also desirable.

**Comment by jakezatecky** [comment]
Offender appears to be the second conditional check at [line 249](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/jsx-sort-props.js#L249).

**Comment by ljharb** [comment]
@jakezatecky if you could provide a PR with a failing test case (or better, a fix!) that would be very appreciated.

**Comment by ljharb** [comment]
Closed by #1632.
